src/common/services/api/cart.service.js

```diff
@@ -82,9 +87,36 @@ class Cart {
       });
   }
 
-  addItem(uri, data){
+  getTotalQuantity() {
+    return this.cortexApiService
+      .get({path: ['carts', this.cortexApiService.scope, 'default']})
+      .map((cart) => {
+        return cart['total-quantity'];
+      });
+  }
+
+  addItem(uri, data) {
     data.quantity = 1;
 
+    if(this.sessionService.getRole() == Roles.public) {
+      return this.getTotalQuantity().mergeMap((total) => {
+        if(total <= 0) {
+          this.$cookies.remove(Sessions.cortex, {path: '/', domain: '.cru.org'});
+          this.$cookies.remove(Sessions.give, {path: '/', domain: '.cru.org'});
+          // Defer til next digest so $cookie.remove propagates.
+          return Observable.from(this.$timeout(angular.noop, 10)).mergeMap(() => {
+            return this.cortexApiService.post({
```

Comment:
http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#static-method-defer might help something. My impression is you provide it with a function that doesn't get called until it is subscribed too. I see now why you have 3 cases. For some reason I thought the last one was always called but the others get returned before it. Do what you think is best. I won't be too picky.